lib/bolt/execution_result.rb

```diff
@@ -47,7 +47,7 @@ def empty
 
     def error_nodes
       result = {}
-      @result_hash.each_pair { |k, v| result[k] = v if v.is_a?(Error) }
+      @result_hash.each_pair { |k, v| result[k] = v if v.is_a?(Puppet::DataTypes::Error) }
```

Comment:
Was this needed?

Reply:
yes it fixed an unreported bug. I moved it to a separate commit. I'll write a test too